go/worker/registration/registration.go

```diff
@@ -228,9 +230,45 @@ func (r *Registration) registerNode(epoch epochtime.EpochTime) error {
 }
 
 func (r *Registration) consensusValidatorHook(n *node.Node) error {
-	addrs, err := r.consensus.GetAddresses()
-	if err != nil {
-		return errors.Wrap(err, "worker/registration: failed to get consensus validator addresses")
+	var addrs []node.ConsensusAddress
+	var err error
+	sentryAddrs := r.workerCommonCfg.SentryAddresses
+	sentryCerts := r.workerCommonCfg.SentryCertificates
+	if len(sentryAddrs) > 0 {
+		// Query sentry nodes for their consensus address(es).
+		for i, sentryAddr := range sentryAddrs {
+			var client *sentryClient.Client
+			client, err = sentryClient.New(r.ctx, &sentryAddr, sentryCerts[i], r.identity)
+			if err != nil {
+				r.logger.Error("failed to create client to a sentry node",
+					"err", err,
+					"sentry_address", sentryAddr,
+				)
+				return err
```

Comment:
returning on errors here means all sentry nodes need to be reachable for this to succeed right? Should instead just `continue` on errors and make sure at least on connection worked?

Reply:
Yes, agreed. I meant to do that, but forgot to change `return`s to `continue`s.

Reply:
Done.